airflow/ti_deps/dep_context.py

```diff
@@ -67,6 +67,8 @@ class DepContext:
     :type ignore_task_deps: bool
     :param ignore_ti_state: Ignore the task instance's previous failure/success
     :type ignore_ti_state: bool
+    :param finished_tasks: A list of all the finished tasks of this run
+    :type finished_tasks: list of airflow.models.TaskInstance
```

Comment:
Ok I checked it, I suggest a change that will solve this and the following comment concerning "dag_run_finished_ti_map" collection. I think it will be both cleaner and more efficient.
 LMK what you think about this direction and will arrange it a bit better if you think its good.